tests/test_preview.py

```diff
@@ -0,0 +1,271 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021-2023, Hewlett Packard Enterprise
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+import pathlib
+from os import path as osp
+
+import pytest
+
+from smartsim import Experiment
+from smartsim._core import Manifest, previewrenderer
+from smartsim._core.config import CONFIG
+from smartsim.settings import RunSettings
+
+
+@pytest.fixture
+def choose_host():
+    def _choose_host(wlmutils, index=0):
+        hosts = wlmutils.get_test_hostlist()
+        if hosts:
+            return hosts[index]
+        else:
+            return None
+
+    return _choose_host
+
+
+def test_experiment_preview(test_dir, wlmutils):
+    """Test correct preview output fields for Experiment preview"""
+    # Prepare entities
+    test_launcher = wlmutils.get_test_launcher()
+    exp_name = "test_prefix"
+    exp = Experiment(exp_name, exp_path=test_dir, launcher=test_launcher)
+
+    # Execute method for template rendering
+    output = previewrenderer.render(exp)
+
+    # Evaluate output
+    summary_lines = output.split("\n")
+    summary_lines = [item.replace("\t", "").strip() for item in summary_lines[-3:]]
+    assert 3 == len(summary_lines)
+    summary_dict = dict(row.split(": ") for row in summary_lines)
```

Comment:
this is clever and i love it, even though i hate having to do the strips to get here...